internal/rpmver/rpmver_test.go

```diff
@@ -0,0 +1,148 @@
+package rpmver
+
+import (
+	"bufio"
+	"encoding"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	gocmp "github.com/google/go-cmp/cmp"
+)
+
+var (
+	_ fmt.Stringer             = (*Version)(nil)
+	_ encoding.TextMarshaler   = (*Version)(nil)
+	_ encoding.TextUnmarshaler = (*Version)(nil)
+)
+
+func TestParse(t *testing.T) {
+	f, err := os.Open("testdata/parse")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	s := bufio.NewScanner(f)
+
+	n := 0
+	for s.Scan() {
+		n++
+		l := s.Text()
+		switch {
+		case len(l) == 0:
+			continue
+		case strings.HasPrefix(l, "#"):
+			continue
+		}
+
+		t.Run(fmt.Sprintf("#%02d", n), func(t *testing.T) {
+			tc := strings.Fields(l)
+			if len(tc) != 6 {
+				t.Fatalf("malformed line: %q (need 6 space-separated fields)", l)
+			}
+			for i := range tc {
+				tc[i] = strings.Trim(tc[i], `"'`)
+			}
+			in, name, epoch, version, release, architecture := tc[0], tc[1], tc[2], tc[3], tc[4], tc[5]
+			// Re-implement the weird epoch behavior.
+			if epoch == "" {
+				epoch = "0"
+			}
+
+			want := Version{
+				Version: version,
+				Release: release,
+				Epoch:   epoch,
+			}
+			if name != "" {
+				want.Name = &name
+			}
+			if architecture != "" {
+				want.Architecture = &architecture
+			}
+
+			got, err := Parse(in)
+			if err != nil {
+				t.Fatalf("%s: %v", in, err)
+			}
+
+			t.Logf("got: %v, want: %v", &got, &want)
+			if !gocmp.Equal(got, want) {
+				t.Fatalf("%s: %v", in, gocmp.Diff(got, want))
+			}
+		})
+	}
+}
+
+func TestRpmvercmp(t *testing.T) {
+	f, err := os.Open("testdata/rpmvercmp")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	s := bufio.NewScanner(f)
+
+	exe, _ := exec.LookPath("rpm")
```

Comment:
should we also ensure we are using a version which would even exist in RHEL? For example, I believe RHEL 9 is using 4.16. Not sure what 10 will use, and not sure what version we are on now, but probably something > 4.16

Reply:
It shouldn't matter, as it's just using the built-in `lua` eval to call the comparison function. [The `rpm` test suite does this](https://github.com/rpm-software-management/rpm/blob/a333eaa3f0fbce0c77cea23015144e13d9c039e2/tests/rpmvercmp.at#L3-L7) (CW: m4), so it should be pretty stable.